assets/variant-list.js

```diff
@@ -159,6 +221,33 @@ class VariantList extends HTMLElement {
       });
   }
 
+  updateButton(quantity) {
+    if (quantity < 0) {
+      if (quantity === -1) {
+        this.querySelector('.variant-list__button-text').innerHTML = window.variantListStrings.itemsRemoved.replace('[quantity]', Math.abs(quantity));
+      } else {
+        this.querySelector('.variant-list__button-text').innerHTML = window.variantListStrings.itemsRemoved.replace('[quantity]', Math.abs(quantity));
+      }
+      this.replaceContent();
+    } else {
+      if (quantity === 1) {
+        this.querySelector('.variant-list__button-text').innerHTML = window.variantListStrings.itemAdded.replace('[quantity]', quantity);
+        this.querySelector('.variant-list__button-icon').classList.remove('hidden');
+      } else {
+        this.querySelector('.variant-list__button-text').innerHTML = window.variantListStrings.itemsAdded.replace('[quantity]', quantity);
+        this.querySelector('.variant-list__button-icon').classList.remove('hidden');
+      }
+      this.replaceContent();
+    }
```

Comment:
Could we use less repetition here?

```suggestion
    const buttonTextElement = this.querySelector('.variant-list__button-text');
    const buttonIconElement = this.querySelector('.variant-list__button-icon');
    const isQuantityNegative = quantity < 0;
    const absQuantity = Math.abs(quantity);
    
    const textTemplate = isQuantityNegative
      ? window.variantListStrings.itemsRemoved
      : (quantity === 1 ? window.variantListStrings.itemAdded : window.variantListStrings.itemsAdded);
    
    buttonTextElement.innerHTML = textTemplate.replace('[quantity]', absQuantity);
    
    if (!isQuantityNegative) {
      buttonIconElement.classList.remove('hidden');
    }
    
    this.replaceContent();
```